scalafix-core/shared/src/main/scala/scalafix/internal/config/DisableConfig.scala

```diff
@@ -1,48 +1,111 @@
 package scalafix
 package internal.config
 
-import metaconfig.ConfDecoder
+import metaconfig._
 import org.langmeta.Symbol
+
 import scalafix.internal.util.SymbolOps
-import metaconfig.annotation.Description
-import metaconfig.annotation.ExampleValue
-import metaconfig.generic
+import metaconfig.annotation.{Description, ExampleValue}
 import metaconfig.generic.Surface
 
+case class UnlessInsideBlock(
+    @Description("The symbol that indicates a 'safe' block.")
+    safeBlock: Symbol.Global,
+    @Description(
+      "The unsafe symbols that are banned unless inside a 'safe' block")
+    symbols: List[CustomMessage[Symbol.Global]])
+
+object UnlessInsideBlock {
+  implicit val surface: Surface[UnlessInsideBlock] =
+    generic.deriveSurface[UnlessInsideBlock]
+  // NOTE(olafur): metaconfig.generic.deriveDecoder requires a default base values.
+  // Here we require all fields to be provided by the user so we write the decoder manually.
+  implicit val reader: ConfDecoder[UnlessInsideBlock] =
+    ConfDecoder.instanceF[UnlessInsideBlock] {
+      case c: Conf.Obj =>
+        (c.get[Symbol.Global]("safeBlock") |@|
+          c.get[List[CustomMessage[Symbol.Global]]]("symbols")).map {
+          case (a, b) => UnlessInsideBlock(a, b)
+        }
+      case _ => Configured.NotOk(ConfError.message("Wrong config format"))
+    }
+}
+
 case class DisableConfig(
-    @Description("The list of symbols to disable.")
-    @ExampleValue(
-      """[
-        |  # With custom message (recommended)
-        |  {
-        |    symbol = "scala.Predef.any2stringadd"
-        |    message = "Use explicit toString before calling +"
-        |  }
-        |  {
-        |    symbol = "scala.Any"
-        |    message = "Explicitly type annotate Any if this is intentional"
-        |  }
-        |  # Without custom message (discouraged)
-        |  "com.Lib.implicitConversion"
-        |]""".stripMargin)
-    symbols: List[CustomMessage[Symbol.Global]] = Nil
+    @Description("The list of symbols to disable only in the actual sources.")
+    @ExampleValue("""
+                    |[
+                    |  {
+                    |    symbol = "scala.Any.asInstanceOf"
+                    |    message = "use pattern-matching instead"
+                    |  }
+                    |]""".stripMargin)
+    symbols: List[CustomMessage[Symbol.Global]] = Nil,
+    @Description(
+      "The list of symbols to disable, also blocks symbols in the generated code")
```

Comment:
List of symbols to disable if inferred. Does not report an error for symbols written explicitly in source code.